integration/sql/src/lib.rs

```diff
@@ -0,0 +1,531 @@
+mod bigquery;
+
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashSet;
+use std::hash::{Hash, Hasher};
+
+pub use bigquery::BigQueryDialect;
+use pyo3::basic::CompareOp;
+use pyo3::exceptions::{PyRuntimeError, PyTypeError};
+use pyo3::prelude::*;
+use sqlparser::ast::{
+    Expr, Ident, Query, Select, SelectItem, SetExpr, Statement, TableAlias, TableFactor, With,
+};
+use sqlparser::dialect::{
+    AnsiDialect, Dialect, GenericDialect, HiveDialect, MsSqlDialect, MySqlDialect,
+    PostgreSqlDialect, SQLiteDialect, SnowflakeDialect,
+};
+use sqlparser::parser::Parser;
+
+pub trait CanonicalDialect: Dialect {
+    fn canonical_name(&mut self, name: String) -> String;
+    fn as_base(&self) -> &dyn Dialect;
+}
+
+impl<T: Dialect> CanonicalDialect for T {
+    fn canonical_name(&mut self, name: String) -> String {
+        if self.is_delimited_identifier_start(name.chars().next().unwrap()) {
```

Comment:
I would try to change unwrap part with some of the alternatives.
This article will help a lot with that https://dmerej.info/blog/post/killing-unwrap/

Reply:
Fixed.